### PR TITLE
[FIX] throw exception if start and end position out of protein

### PR DESCRIPTION
**Pull request by timosachsenberg**
boundaries in coverage calculation

### CODE REVIEW

**Comment by mwalzer** [comment]
nice

File: src/openms/source/FORMAT/FeatureXMLFile.cpp [hunk 1]
```diff
@@ -1061,33 +1061,42 @@ namespace OpenMS
     // write peptide hits
     for (Size j = 0; j < id.getHits().size(); ++j)
     {
+      const PeptideHit & h = id.getHits()[j];
```
Comment (hendrikweisser) [inline src/openms/source/FORMAT/FeatureXMLFile.cpp]: No space before "&".

File: src/openms/source/FORMAT/FeatureXMLFile.cpp [hunk 2]
```diff
@@ -1061,33 +1061,42 @@ namespace OpenMS
...
       os << indent << "\t<PeptideHit";
-      os << " score=\"" << id.getHits()[j].getScore() << "\"";
-      os << " sequence=\"" << id.getHits()[j].getSequence() << "\"";
-      os << " charge=\"" << id.getHits()[j].getCharge() << "\"";
+      os << " score=\"" << h.getScore() << "\"";
+      os << " sequence=\"" << h.getSequence() << "\"";
+      os << " charge=\"" << h.getCharge() << "\"";
 
-      vector<PeptideEvidence> pes = id.getHits()[j].getPeptideEvidences();
+      const vector<PeptideEvidence> & pes = id.getHits()[j].getPeptideEvidences();
```
Comment (hendrikweisser) [inline src/openms/source/FORMAT/FeatureXMLFile.cpp]: No space before "&".

Comment (timosachsenberg) [inline src/openms/source/FORMAT/FeatureXMLFile.cpp]: ok can do.  Might be good to add this once and for all the cpplint script.

**Pull request merged**